lib/utils/dates.ts

```diff
@@ -46,40 +46,48 @@ export const isValidDate = (maybeDate: unknown): boolean => {
   return false;
 };
 
+/**
+ * Get total number of days in a particular month
+ *
+ * @param      {Date}    date    Date of the month that should be considered
+ * @return     {number}  Total number of dates in the particular month
+ */
 export const daysInMonth = (date: Date): number => {
-  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
+  return new Date(date.getUTCFullYear(), date.getUTCMonth() + 1, 0).getUTCDate();
 };
 
 /**
  * For a given date, get the ISO week number
- * @param date
+ * @param    date
+ * @returns  ISO week number of the specified date
  */
 export const getWeekNumber = (date: Date): number => {
-  const newYear = new Date(date.getFullYear(), 0, 1);
-  let day = newYear.getDay(); // The day of week the year begins on
-  day = day >= 0 ? day : day + 7;
-  const dayNum =
-    Math.floor(
-      (date.getTime() - newYear.getTime() - (date.getTimezoneOffset() - newYear.getTimezoneOffset()) * 60000) /
-        86400000,
-    ) + 1;
-  let weekNum;
-  // If the year starts before the middle of a week
-  if (day < 4) {
-    weekNum = Math.floor((dayNum + day - 1) / 7) + 1;
-    if (weekNum > 52) {
-      const nYear = new Date(date.getFullYear() + 1, 0, 1);
-      let nDay = nYear.getDay();
-      nDay = nDay >= 0 ? nDay : nDay + 7;
-      // If the next year starts before the middle of the week, it is week #1 of that year
-      weekNum = nDay < 4 ? 1 : 53;
-    }
-  } else {
-    weekNum = Math.floor((dayNum + day - 1) / 7);
-  }
-  return weekNum;
+  const d = new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
+  d.setUTCDate(d.getUTCDate() + 4 - (d.getUTCDay() || 7));
+  return Math.ceil(((d.getTime() - new Date(Date.UTC(d.getUTCFullYear(), 0, 1)).getTime()) / 864e5 + 1) / 7)
```

Comment:
👍

Reply:
Well, I am not that good that algorithm. I sourced it from [SO](https://stackoverflow.com/a/6117889/3408342). :wink: